libs/cardano-ledger-core/src/Cardano/Ledger/Plutus/ExUnits.hs

```diff
@@ -163,9 +165,34 @@ instance NoThunks Prices
 
 instance NFData Prices
 
-instance ToJSON Prices
-
-instance FromJSON Prices
+instance ToJSON Prices where
+  toJSON Prices {prSteps, prMem} =
+    -- We cannot round-trip via NonNegativeInterval, so we go via Rational
+    object
+      [ "priceSteps" .= toRationalJSON (unboundRational prSteps)
+      , "priceMemory" .= toRationalJSON (unboundRational prMem)
+      ]
+    where
+      toRationalJSON :: Rational -> Value
+      toRationalJSON r =
+        case fromRationalRepetendLimited 20 r of
+          Right (s, Nothing) -> toJSON s
+          _ -> toJSON r
```

Comment:
Yes we can roundtrip. This has been fixed recently in #4058 
```suggestion
    object
      [ "priceSteps" .= prSteps
      , "priceMemory" .= prMem
      ]
```

Reply:
Thanks, I didn't notice because I was working on top of and old commit